### ISSUE TITLE
Cannot Install Edge with Bundler

### ISSUE THREAD

**Issue by reckerswartz** [rails/rails#36112]
### Steps to reproduce

Setp 1:
new **Gemfile** with

```
source 'https://rubygems.org'

gem 'rails', git: 'https://github.com/rails/rails.git'
gem 'arel', git: 'https://github.com/rails/arel.git'
```
Setp 2:
`bundle install`

setup 3:
`rails new . --dev -T -d postgresql`

### Expected behavior
Initial clean Edge Rails App

### Actual behavior
```
Bundler could not find compatible versions for gem "activesupport":
  In Gemfile:
    rails was resolved to 6.1.0.alpha, which depends on
      activesupport (= 6.1.0.alpha)

    webpacker was resolved to 4.0.2, which depends on
      activesupport (>= 4.2)

Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    rails was resolved to 6.1.0.alpha, which depends on
      railties (= 6.1.0.alpha)

    sass-rails (~> 5) was resolved to 5.0.4, which depends on
      railties (>= 4.0.0, < 5.0)

    web-console was resolved to 3.7.0, which depends on
      railties (>= 6.0.0.a)

    webpacker was resolved to 4.0.2, which depends on
      railties (>= 4.2)
         run  bundle binstubs bundler
The git source https://github.com/rails/webpacker.git is not yet checked out. Please run `bundle install` before trying to start your
application
       rails  webpacker:install
The git source https://github.com/rails/webpacker.git is not yet checked out. Please run `bundle install` before trying to start your application
```

### System configuration
**Rails version**: 2.6.2

**Ruby version**: Edge

**Comment by rafaelfranca** [comment]
Which version of Rails did you use the generate the application?

**Comment by reckerswartz** [comment]
@rafaelfranca  logs
```
rails new . --dev -T -d postgresql
       exist  
      create  README.md
      create  Rakefile
      create  .ruby-version
      create  config.ru
      create  .gitignore
    conflict  Gemfile
Overwrite /home/recker/workspace/ex/test/aventumx_client_web/Gemfile? (enter "h" for help) [Ynaqdhm] y
       force  Gemfile
         run  git init from "."
Initialized empty Git repository in /home/recker/workspace/ex/test/aventumx_client_web/.git/
      create  package.json
      create  app
      create  app/assets/config/manifest.js
      create  app/assets/stylesheets/application.css
      create  app/channels/application_cable/channel.rb
      create  app/channels/application_cable/connection.rb
      create  app/controllers/application_controller.rb
      create  app/helpers/application_helper.rb
      create  app/javascript/channels/consumer.js
      create  app/javascript/channels/index.js
      create  app/javascript/packs/application.js
      create  app/jobs/application_job.rb
      create  app/mailers/application_mailer.rb
      create  app/models/application_record.rb
      create  app/views/layouts/application.html.erb
      create  app/views/layouts/mailer.html.erb
      create  app/views/layouts/mailer.text.erb
      create  app/assets/images/.keep
      create  app/controllers/concerns/.keep
      create  app/models/concerns/.keep
      create  bin
      create  bin/rails
      create  bin/rake
      create  bin/setup
      create  bin/yarn
      create  config
      create  config/routes.rb
      create  config/application.rb
      create  config/environment.rb
      create  config/cable.yml
      create  config/puma.rb
      create  config/storage.yml
      create  config/environments
      create  config/environments/development.rb
      create  config/environments/production.rb
      create  config/environments/test.rb
      create  config/initializers
      create  config/initializers/application_controller_renderer.rb
      create  config/initializers/assets.rb
      create  config/initializers/backtrace_silencers.rb
      create  config/initializers/content_security_policy.rb
      create  config/initializers/cookies_serializer.rb
      create  config/initializers/cors.rb
      create  config/initializers/filter_parameter_logging.rb
      create  config/initializers/inflections.rb
      create  config/initializers/mime_types.rb
      create  config/initializers/new_framework_defaults_6_0.rb
      create  config/initializers/wrap_parameters.rb
      create  config/locales
      create  config/locales/en.yml
      create  config/master.key
      append  .gitignore
      create  config/boot.rb
      create  config/database.yml
      create  db
      create  db/seeds.rb
      create  lib
      create  lib/tasks
      create  lib/tasks/.keep
      create  lib/assets
      create  lib/assets/.keep
      create  log
      create  log/.keep
      create  public
      create  public/404.html
      create  public/422.html
      create  public/500.html
      create  public/apple-touch-icon-precomposed.png
      create  public/apple-touch-icon.png
      create  public/favicon.ico
      create  public/robots.txt
      create  tmp
      create  tmp/.keep
      create  tmp/cache
      create  tmp/cache/assets
      create  vendor
      create  vendor/.keep
      create  storage
      create  storage/.keep
      create  tmp/storage
      create  tmp/storage/.keep
      remove  config/initializers/cors.rb
      remove  config/initializers/new_framework_defaults_6_0.rb
         run  bundle install
Fetching https://github.com/rails/webpacker.git
Fetching https://github.com/rails/web-console.git
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
Fetching gem metadata from https://rubygems.org/............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "activesupport":
  In Gemfile:
    rails was resolved to 6.1.0.alpha, which depends on
      activesupport (= 6.1.0.alpha)

    webpacker was resolved to 4.0.2, which depends on
      activesupport (>= 4.2)

Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    rails was resolved to 6.1.0.alpha, which depends on
      railties (= 6.1.0.alpha)

    sass-rails (~> 5) was resolved to 5.0.4, which depends on
      railties (>= 4.0.0, < 5.0)

    web-console was resolved to 3.7.0, which depends on
      railties (>= 6.0.0.a)

    webpacker was resolved to 4.0.2, which depends on
      railties (>= 4.2)
         run  bundle binstubs bundler
The git source https://github.com/rails/webpacker.git is not yet checked out. Please run `bundle install` before trying to start your
application
       rails  webpacker:install
The git source https://github.com/rails/webpacker.git is no
````

**Comment by itsWill** [comment]
This seems like a related issue to this previously closed one: https://github.com/rails/rails/issues/36084
